testing/testcafe/tests/dataGrid/editing.ts

```diff
@@ -2291,3 +2293,66 @@ test('Popup EditForm screenshot', async (t) => {
     })();
   });
 });
+
+[
+  {
+    theme: 'material.blue.light',
+    useIcons: true,
+  },
+  {
+    theme: 'generic.light',
+    useIcons: true,
+  },
+  {
+    theme: 'material.blue.light',
+    useIcons: false,
+  },
+  {
+    theme: 'generic.light',
+    useIcons: false,
+  },
+].forEach(({ theme, useIcons }) => {
+  // T1179114
+  test(`The disabled state should be correct for a custom button when given as a SVG image (${theme})`, async (t) => {
+    const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+    const dataGrid = new DataGrid('#container');
+
+    await t
+      .expect(await takeScreenshot(`T1179114-grid-edit-custom-button-in-${theme.split('.')[0]}-theme-when-useicons-is-${useIcons}.png`, dataGrid.element))
+      .ok()
+      .expect(compareResults.isValid())
+      .ok(compareResults.errorMessages());
+  }).before(async () => {
+    await changeTheme(theme);
+
+    return createWidget('dxDataGrid', {
+      height: 150,
+      width: 600,
+      dataSource: [{
+        Id: 0,
+        name: 'test',
+      }],
+      keyExpr: 'Id',
+      editing: {
+        mode: 'row',
+        allowUpdating: true,
+        allowDeleting: true,
+        useIcons,
+      },
+      columns: ['Id', 'name', {
+        type: 'buttons',
+        buttons: ['edit',
+          {
+            name: 'delete',
+            disabled: true,
+          },
+          {
+            icon: encodedIcon,
+            disabled: true,
+          }],
+      }],
```

Comment:
Let's change the buttons here to make the test and screenshots much obvious as possible.

I suggest adding pairs of the same disabled and non-disabled buttons to see the difference on an etalon screenshot.

E.g.
```suggestion
      columns: ['Id', 'name', {
        type: 'buttons',
        buttons: [
          {
            name: 'delete',
            disabled: false,
          },
          {
            name: 'delete',
            disabled: true,
          },
          {
            icon: encodedIcon,
            disabled: false,
          },
          {
            icon: encodedIcon,
            disabled: true,
          }
        ],
      }],
```